gpiozero/output_devices.py

```diff
@@ -300,7 +299,10 @@ def __init__(self, pin=None, active_high=True, initial_value=0, frequency=100):
         self._controller = None
         if not 0 <= initial_value <= 1:
             raise OutputDeviceBadValue("initial_value must be between 0 and 1")
-        super(PWMOutputDevice, self).__init__(pin, active_high)
+        if initial_value in (0, 1):
+            super(PWMOutputDevice, self).__init__(pin, active_high, initial_value=initial_value)
```

Comment:
> What do you mean?

The way it works currently, is if you create a PWMOutputDevice with an intial_value of e.g. 1.0, then it'll first call the super() init-method in OutputDevice, which has a default initial_value of False. So the pin will get set to (digital) output-mode, with a value of 0. Then back in the PWMOutputDevice init-method we set the frequency property to the required value, which 'turns on' PWM mode, and again sets the value to 0.0. And finally in the PWMOutputDevice init-method we set the desired value, which eventually gets our pin operating at the correct value.

Ideally, the pin wouldn't enter 'digital' output mode, and wouldn't have a PWM-value of 0, but would immediately switch from 'input' to outputting the desired PWM-value.
(Yes, for all practical limitations there's no *actual* difference, but maybe there's a theoretical PWM device which should *never* have it's output driven to 0, no matter for how short a time-period? Possibly I'm just being overly-fussy.)